test/e2e/apimachinery/watch.go

```diff
@@ -314,6 +316,49 @@ var _ = SIGDescribe("Watchers", func() {
 		expectEvent(testWatch, watch.Modified, testConfigMapThirdUpdate)
 		expectEvent(testWatch, watch.Deleted, nil)
 	})
+
+	/*
+		    Testname: watch-consistency
+		    Description: Ensure that concurrent watches are consistent with each other by initiating an additional watch
+		    for events recieved from the first watch, initiated at the resource version of the event, and checking that all
+		    resource versions of all events match. Events are produced from writes on a background goroutine.
+	*/
+	framework.ConformanceIt("should recieve events on concurrent watches in same order", func() {
+		c := f.ClientSet
+		ns := f.Namespace.Name
+
+		iterations := 100
+
+		By("starting a background goroutine to produce watch events")
+		donec := make(chan struct{})
+		stopc := make(chan struct{})
+		go func() {
+			defer GinkgoRecover()
+			produceConfigMapEvents(f, stopc, 5*time.Millisecond)
+			close(donec)
+		}()
+
+		By("creating watches starting from each resource version of the events produced and verifying they all recieve resource versions in the same order")
+		wcs := []watch.Interface{}
+		resourceVersion := "0"
+		for i := 0; i < iterations; i++ {
+			wc, err := c.CoreV1().ConfigMaps(ns).Watch(metav1.ListOptions{ResourceVersion: resourceVersion})
+			Expect(err).NotTo(HaveOccurred())
+			wcs = append(wcs, wc)
+			resourceVersion = waitForNextConfigMapEvent(wcs[0]).ResourceVersion
+			for _, wc := range wcs[1:] {
```

Comment:
A bit to my surprise, this didn't speed things up.  I'm guessing that the buffered channels `<-watch.ResultChan()` in `waitForNextConfigMapEvent()` are sufficient since the watches are initiated in the outer for loop.

Reply:
Interesting.